### ISSUE TITLE
Broadcast not working

### ISSUE THREAD

**Issue by Mark-Aerobat** [dannagle/PacketSender#288]
#### What OS?
iPhone and iPad IOS 15.3.1

#### Description of issue
UDP broadcast using IOS app does not work.  Only unicast works.

UPD broadcast to the same address works in the Windows program

![1 - Windows UDP Broadcast Works](https://user-images.githubusercontent.com/101292238/157536125-152dff2d-71cc-4d08-81ff-58e07b25381d.jpg)

![2 - IOS UDP Unicast Works](https://user-images.githubusercontent.com/101292238/157536140-5a890565-b9f0-4610-9eff-7ae7399f454d.jpg)

![3 - IOS UDP Broadcast not working](https://user-images.githubusercontent.com/101292238/157536148-5547f52a-9134-48f1-a112-23fdeda61079.jpg)

**Comment by dannagle** [comment]
I dug into this a bit. It seems Apple blocks broadcast and multicast. However, there may be a way if I can have the app request the correct special permission.

**Comment by Mark-Aerobat** [comment]
Thanks for looking into it.

**Comment by Mark-Aerobat** [comment]
Any progress on this issue?  Were you able to request the correct special permission from Apple?

**Comment by dannagle** [comment]
This fell off my radar. I've been putting out so many bigger fires. I'll put this higher on my list.

**Comment by dannagle** [comment]
Just an FYI... I've requested the entitlement for my app from Apple. Apparently, this feature requires a special request which is why I have been having trouble getting it to work.

**Comment by Mark-Aerobat** [comment]
Thanks for following up and for requesting the entitlement.  It will be a useful feature.

**Comment by Mark-Aerobat** [comment]
Any news from Apple concerning your entitlement for UDP broadcast?

**Comment by dannagle** [comment]
Yes, it got approved last week. I am preparing for a conference next week, and then I can add the fix to the app. Thanks for asking. I am marking this thread unread in my inbox.

**Comment by Mark-Aerobat** [comment]
When you have a chance can you update the app to include UDP broadcast. It would be a good feature to have for my current project.  Thanks.

**Comment by Mark-Aerobat** [comment]
Hope that all is well.  Are you still planning to add the UDP broadcast feature to the iOS app?  It would be a great feature to have in the app and it would also be very useful for my current project.  Let me know what your plans are.

**Comment by dannagle** [comment]
I am sorry. You have been super patient. After a lot of thought, I decided to remove the app from sale until I actually have the time and budget to support it properly. I keep waiting for my schedule to ease, and it simply never does.

Beyond just broadcast/multicast support, the app needs a lot of work to reach better feature parity with the desktop. The Android version was removed too.

**Comment by Mark-Aerobat** [comment]
I understand.  Thanks for letting me know.

**Comment by dannagle** [comment]
The reworked / republished iOS app should have broadcast support.

**Comment by Mark-Aerobat** [comment]
Thanks for updating the IOS app to include UDP broadcast.  

I tried to send a broadcast packet but it did not work.  Unicast to a single IP works as expected but the broadcast packet is not sent from IOS.  The exact same UDP broadcast configuration on the desktop app works perfectly.

Did you receive the necessary entitlement from Apple to use UDP broadcast in your IOS app?

**Comment by dannagle** [comment]
Things seemed to be wired correctly.  I got the entitlement enabled in Apple. I checked the archive I uploaded, and it is set in there too. 

However, I am getting the same results as you.  general broadcast and subnet broadcast not appearing. What is possible is my test router is consumer grade and could be blocking it. Broadcast over WiFi is a difficult test to set up. 

I'll try to get a better test going soon.

**Comment by Mark-Aerobat** [comment]
I did a test with the same WiFi router from a PC and IOS.  It works from the PC but does not work in IOS.  It might be worth trying the same test on your side to eliminate any questions about the router.

**Comment by Mark-Aerobat** [comment]
Thanks for continuing to look into it.  It should be very useful once it is up and running.

**Comment by Mark-Aerobat** [comment]
Thanks for the update. I’m glad you found the underlying cause, even though it’s frustrating that it comes down to Apple policy rather than code. I really appreciate you trying to avoid a “new app/new App ID” solution that would impact existing customers. Hopefully Apple can help unblock this.

**Comment by Mark-Aerobat** [comment]
If a new app with broadcast capability is developed, one way to avoid disappointing users who already purchased the current version would be to price the broadcast version slightly higher—perhaps about $1 more. This would position it as an enhanced version of the standard app.

The small price increase would be easy to justify given the additional work required to implement broadcast functionality. Users who need the feature could simply purchase the enhanced version, while others could continue using the standard app.

**Comment by Mark-Aerobat** [comment]
Have you heard back from Apple about applying the entitlement to the wildcard App ID for the existing app?

**Comment by dannagle** [comment]
No. I am going to file a fresh request.

**Comment by Mark-Aerobat** [comment]
Thanks for your efforts to get broadcast working.  Do you have any news on your fresh request?
